src/Tmds.DBus/PropertyAttribute.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using Tmds.DBus.CodeGen;
+
+namespace Tmds.DBus
+{
+    /// <summary>
+    /// Overrides how the property is handled.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Field)]
+    public class PropertyAttribute : Attribute
+    {
+        /// <summary>
+        /// If not null, used to override the autogenerated name.
+        /// </summary>
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Specifies the mutability of the property.
+        /// </summary>
+        public PropertyAccess Access { get; set; } = PropertyAccess.ReadWrite;
+
+        /// <summary>
+        /// If not null, used to override the type used in the signature.
+        /// </summary>
+        public Type Type { get; set; }
```

Comment:
We need to (de)serialize the field type, so I don't think a user should override this.

Reply:
Removed.